libs/unity/library/Runtime/Scripts/NativeRender/NativeVideoRenderer.cs

```diff
@@ -42,12 +47,15 @@ private void OnDestroy()
 
         private void Update()
         {
-            if (_i420aFrameQueue != null)
+            if (_nativeVideo != null && (_textureY.width != dirtyWidth || _textureY.height != dirtyHeight))
             {
-                if (_i420aFrameQueue.TryDequeue(out I420AVideoFrameStorage frame) && _nativeVideo == null)
-                {
-                    StartNativeRendering((int)frame.Width, (int)frame.Height);
-                }
+                CreateEmptyVideoTextures(dirtyWidth, dirtyHeight, 128);
+                _nativeVideo.UpdateTextures(GetTextureDescArray());
+            }
+            
+            if (_nativeVideo == null && _i420AFrameQueue.TryDequeue(out I420AVideoFrameStorage frame))
+            {
+                StartNativeRendering((int)frame.Width, (int)frame.Height);
```

Comment:
We're discarding the frame here?

Reply:
Can't we start the renderer AND also enqueue the frame, to not lose it? Or is it not worth the complication?

Reply:
The issue is that unity must know the texture resolution to then create the texture and send it to the native side, and that texture creation has to be done from the unity main thread. 

So it gets kind of odd as you'd have to wait for the unity thread to respond after the first frame.

Reply:
I could make it just rely on the UpdateTextures method to resize it, would make the C# simpler.